plutus-contract/src/Plutus/Contract/Test/ContractModel/Internal.hs

```diff
@@ -586,24 +689,19 @@ handle handles key =
         Just (WalletContractHandle _ h) -> h
         Nothing                         -> error $ "handle: No handle for " ++ show key
 
+type AssetMap = Map AssetKey (Map String AssetClass)
+
 -- | The `EmulatorTrace` monad does not let you get the result of a computation out, but the way
 --   "Test.QuickCheck.Monadic" is set up requires you to provide a function @m Property -> Property@.
 --   This means that we can't use `EmulatorTrace` as the action monad in the `StateModel`. Instead
 --   we use a state monad that builds up an `EmulatorTrace` computation to be executed at the end
 --   (by `finalChecks`). We also need access to the contract handles, so what we are building is a
 --   function from the handles to an emulator trace computation returning potentially updated
 --   handles.
-type ContractMonad state = State.State (ContractMonadState state)
-
-data ContractMonadState state = ContractMonadState (EmulatorAction state) [ContractInstanceSpec state]
-
-instance Semigroup (ContractMonadState state) where
-    ContractMonadState f xs <> ContractMonadState g ys = ContractMonadState (f <> g) (xs <> ys)
-
-instance Monoid (ContractMonadState state) where
-    mempty = ContractMonadState mempty mempty
 
-newtype EmulatorAction state = EmulatorAction { runEmulatorAction :: Handles state -> EmulatorTrace (Handles state) }
+-- TODO: Refactor this(!)
```

Comment:
Just checking that you didn't leave this here by accident.

Reply:
Unfortunately this is here on purpose. Our ambition is to refactor parts of the `ContractModel` machinery but that should come in a future PR when we get around to it - right now it's low priority.